id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java

```diff
@@ -187,8 +235,14 @@ public Map<AllowedKycDto, Object> prepareSharableAttributes(IdResponseDTO idResp
 		Set<AllowedKycDto> sharableAttributeDemographicKeySet = new HashSet<>();
 		Set<AllowedKycDto> sharableAttributeBiometricKeySet = new HashSet<>();
 			List<String> userRequestedAttributes = credentialServiceRequestDto.getSharableAttributes();
-			Map<String, Object> additionalData = credentialServiceRequestDto.getAdditionalData();
+			Map<String, Object> additionalData = credentialServiceRequestDto.getAdditionalData();			
+			if (sharableAttributeList != null && userRequestedAttributes != null) {
+				userRequestedSharableAttributesList = sharableAttributeList.stream()

```

Comment:
As discussed already, Check the intersection of list by iterating userRequestedAttributes.